Source/WebCore/rendering/TextBoxPainter.cpp

```diff
@@ -79,17 +79,17 @@ TextBoxPainter::TextBoxPainter(const LayoutIntegration::InlineContent& inlineCon
     }())
     , m_isPrinting(m_document.printing())
     , m_haveSelection(computeHaveSelection())
-    , m_containsComposition(m_renderer.textNode() && m_renderer.frame().editor().compositionNode() == m_renderer.textNode())
-    , m_useCustomUnderlines(m_containsComposition && m_renderer.frame().editor().compositionUsesCustomUnderlines())
     , m_emphasisMarkExistsAndIsAbove(RenderText::emphasisMarkExistsAndIsAbove(m_renderer, m_style))
 {
     ASSERT(paintInfo.phase == PaintPhase::Foreground || paintInfo.phase == PaintPhase::Selection || paintInfo.phase == PaintPhase::TextClip || paintInfo.phase == PaintPhase::EventRegion || paintInfo.phase == PaintPhase::Accessibility);
-}
 
-TextBoxPainter::~TextBoxPainter()
-{
+    auto& editor = m_renderer.frame().editor();
+    m_containsComposition = m_renderer.textNode() && editor.compositionNode() == m_renderer.textNode();
+    m_useCustomUnderlines = m_containsComposition && editor.compositionUsesCustomUnderlines();
```

Comment:
I doubt this saves much work tbh, the most expensive bit of LocalFrame::editor() is `ensureEditor()`, which should only be invoked on the first member of this class.

Reply:
The cost is jumping from RenderObject to Document to Frame twice.